src/providers/Cart/index.js

```diff
@@ -0,0 +1,85 @@
+import React, {createContext, useContext, useEffect} from "react";
+import { GENERAL, OPERATIONS } from "../../utils/constants";
+
+const CartContext = createContext()
+
+export const useCartContext = () => {
+    const context = useContext(CartContext)
+
+    if (!context) {
+        throw new Error(`CartContext provider is not found`);
+    }
+
+    return context;    
+};
+
+export const CartProvider = ({children}) => {
+    const [cartItems, setCartItems] = React.useState(GENERAL.PRODUCT_EMPTY)
+    const [itemsInCart, setItemsInCart] = React.useState([])
+    const [currentProduct, setCurrentProduct] = React.useState({})
+    const [total, setTotal] = React.useState(0)
+    
+
+    useEffect(() => {
+        if (Object.keys(currentProduct).length){
+            setItemsInCart(prevItems => {
+                
+                const exist = [...prevItems].find((element) => {
+                    return element.id === currentProduct.id
+                })
```

Comment:
al usar el `find` no es necesario crear un array nuevo, `find` no modifica el arreglo original:

```suggestion
                const exist = prevItems.find((element) => {
                    return element.id === currentProduct.id
                })
```